### PR TITLE
Release script for the new release process

### PR DESCRIPTION
**Pull request by bozzzzo**

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -1,78 +1,100 @@
-# Releasing Quark
+# Releasing Quark #
 
-## Before releasing...
+The release workflow has the following steps, which are scripted.  All
+commands below take additional options, see `./release --help` for
+details.  Publishing artefacts to pypi requires write permission to
```
Comment (brikelly) [inline RELEASE.md]: "Artefacts" is misspelled throughout; it should be "artifacts".

File: RELEASE.md [hunk 2]
```diff
@@ -1,78 +1,100 @@
...
->
-> The overall idea is to make sure the process of a) understanding class of problems quark solves, b) trying out the examples we provide, and c) adapting our examples to fit their own specific problems is as streamlined as possible for prospective users.
+Since the release script is part of the repository, and release
+involves switching branches, the release script may not be stable
+across the branches. The best approach is to use two clones of the
+repo, one to provide the script and one to be manipulated by the
+script.
 
-## Prerequisites for every release:
+    # release = $(./release freeze)
```
Comment (brikelly) [inline RELEASE.md]: Are you trying to use the "#" character to denote a shell command, or as markup? It's not rendering as markup, and it's confusing to see it used on the same line as "$" later (which is more typically used to denote shell commands).

**Comment by bozzzzo** [comment]
A good way to test all this:
```
cd /tmp
git clone --mirror git@github.com:datawire/quark.git        # git fetch will refresh it

# the following steps can then be quickly retried
git clone --bare quark.git temp.git
git clone temp.git test
cd test
# git tag -d $(git tag --list dev-0.5.18?)   # pretend some develop builds were not tagged yet
eval $(./release freeze)
quark_release poll-dev-status --tag-dev-builds
quark_release prepare-release --prod
quark_release push-release                   # push is safe in here as it will push to temp.git
cd ../temp.git
# inspect changes
```

File: RELEASE.md [hunk 3]
```diff
@@ -1,78 +1,119 @@
...
+manually as part of the release process.
 
-1. Review the current release metadata and determine the appropriate
-   version for the next release:
+To query the CI system for new builds of the develop branch and tag the successful ones
 
-   ```
-   ./release version
-   ```
+    $ release poll-dev-status --tag-dev-builds
```
Comment (rhs) [inline RELEASE.md]: Should this actually be ``releease`` or ``quark_release``?

Comment (bozzzzo) [inline RELEASE.md]: strictly speaking, it should. Will fix up.

**Comment by bozzzzo** [comment]
Merged manually

**Pull request closed without merging**